src/Microsoft.AspNetCore.Server.Kestrel.Https/HttpsConnectionAdapterOptions.cs

```diff
@@ -51,5 +57,21 @@ public HttpsConnectionAdapterOptions()
         /// Specifies whether the certificate revocation list is checked during authentication.
         /// </summary>
         public bool CheckCertificateRevocation { get; set; }
+
+        /// <summary>
+        /// Specifies the maximum amount of time allowed for the TLS/SSL handshake.
+        /// </summary>
+        public TimeSpan HandshakeTimeout
```

Comment:
Hmm have we added public API surface in patches before?

Reply:
Not that I know of, but I think we should keep this property to mitigate the risk of this patch.

Reply:
Can we add a quirks switch instead that reverts to the 2.0.0 RTM behavior?